cli_helpers/tabular_output/preprocessors.py

```diff
@@ -83,6 +83,7 @@ def align_decimals(data, headers, column_types=(), **_):
 
     """
     pointpos = len(headers) * [0]
+    data = list(data)
```

Comment:
So, right below this we do `for row in data:`, which should work on an iterator, right? If not, we'll need to update the other preprocessors as well, too.